src/ngraph/codegen/compiler.cpp

```diff
@@ -511,3 +518,25 @@ string codegen::CompilerCore::find_header_version(const string& path)
     }
     return rc;
 }
+
+string codegen::CompilerCore::find_os_specific_path(const string& path)
+{
+    string rc;
+    auto f = [&](const std::string& file, bool is_dir) {
+        if (is_dir)
+        {
+            const string prefix = "x86_64-";
+            const string suffix = "-linux";
+            string path = file_util::get_file_name(file);
+            if (path.size() > prefix.size() && path.size() > suffix.size() &&
```

Comment:
Should be `path.size() > prefix.size() + suffix.size()` or your substring below could end up with the start after the end.